magento2-nginx/fpm-7.0/usr/local/share/container/baseimage-30.sh

```diff
@@ -12,17 +14,25 @@ do_development_start() {
   do_magento2_development_start
 }
 
+alias_function do_templating do_magento2_templating_inner
+do_templating() {
+  mkdir -p /home/build/.hem/gems/
+  chown -R build:build /home/build/.hem/
```

Comment:
you can fix this issue by doing `useradd build --create-home`, then Hem will be able to create the directory